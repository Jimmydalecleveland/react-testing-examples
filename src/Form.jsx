import React from 'react';
import { Formik } from 'formik';

const Basic = ({ subby }) => (
  <div>
    <h1>Anywhere in your app!</h1>
    <Formik
      initialValues={{ fullname: '' }}
      onSubmit={subby}
    >
      {({
        values,
        errors,
        touched,
        handleChange,
        handleBlur,
        handleSubmit,
        isSubmitting,
        /* and other goodies */
      }) => (
        <form onSubmit={handleSubmit}>
          <label htmlFor="fullname">
          Full Name
          <input
            id="fullname"
            type="text"
            name="fullname"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.fullname}
            required
          />

          </label>
          {errors.fullname && touched.fullname}
          <button type="submit" disabled={isSubmitting}>
            Submit
          </button>
        </form>
      )}
    </Formik>
  </div>
);

export default Basic;