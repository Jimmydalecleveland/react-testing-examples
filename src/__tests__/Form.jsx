import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'

import Form from '../Form'

describe('Form', () => {
  test('should render', async () => {
    const mockSubmit = jest.fn()
    const { container, getByText, getByLabelText } = render(
      <Form subby={mockSubmit}></Form>
    )

    expect(container).toMatchSnapshot()
    const submitButton = getByText(/submit/i)
    const nameField = getByLabelText(/full name/i)
    expect(nameField).toBeRequired()
    fireEvent.click(submitButton)
    await wait(() => {
      expect(mockSubmit).toHaveBeenCalledTimes(1)
    })
  })
})
