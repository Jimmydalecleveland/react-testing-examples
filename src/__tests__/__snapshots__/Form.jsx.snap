// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render 1`] = `
<div>
  <div>
    <h1>
      Anywhere in your app!
    </h1>
    <form>
      <label
        for="fullname"
      >
        Full Name
        <input
          id="fullname"
          name="fullname"
          required=""
          type="text"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
